anta/tests/logging.py

```diff
@@ -0,0 +1,297 @@
+"""
+Test functions related to the EOS various logging settings
+"""
+from __future__ import annotations
+
+import logging
+import re
+from typing import Dict, List, Optional, cast
+
+from anta.models import AntaTest, AntaTestCommand
+
+logger = logging.getLogger(__name__)
+
+
+class VerifyLoggingPersistent(AntaTest):
+    """
+    Verifies if logging persistent is enabled and logs are saved in flash.
+
+    Expected Results:
+        * success: The test will pass if logging persistent is enabled and logs are in flash.
+        * failure: The test will fail if logging persistent is disabled or no logs are saved in flash.
+    """
+
+    name = "VerifyLoggingPersistent"
+    description = "Verifies if logging persistent is enabled and logs are saved in flash."
+    categories = ["logging"]
+    commands = [
+        AntaTestCommand(command="show logging | awk 'NR==4 {print; exit}'", ofmt="text"),
+        AntaTestCommand(command="dir flash:/persist/messages", ofmt="text"),
+    ]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """
+        Run VerifyLoggingPersistent validation.
+        """
+        self.logger.debug(f"self.instance_commands is: {self.instance_commands}")
+        log_output = cast(str, self.instance_commands[0].output)
+        dir_flash_output = cast(str, self.instance_commands[1].output)
+        self.logger.debug(f"dataset of log_output command is: {log_output}")
+        self.logger.debug(f"dataset of dir_flash_output command is: {dir_flash_output}")
+
+        pattern = r"-rw-\s+(\d+)"
+        match = re.search(pattern, dir_flash_output)
+
+        if match and "disabled" not in log_output and int(match.group(1)) > 0:
+            self.result.is_success()
+        else:
+            self.result.is_failure("Persistent logging is disabled or no logs are saved in flash")
```

Comment:
so I would rather write it like this:
with an optional return if you want to stop after finding out that persistent logging is disabled.

Random question - is your test not able to catch false positive if the log file is actually 0 in size because no log has been captured yet?

```suggestion
        self.result.is_success()
        if "disabled" in log_output:
            self.result.is_failure("Persistent logging is disabled!")
            # return
        if not match or int(match.group(1)) != 0:
            self.result.is_failure("No persistent logs are saved in flash")
```

Reply:
No because the second you enable `logging persistent` logs are generated in `flash:/persist/messages` ;

```
NW-CORE(config)#more flash:/persist/messages
2023-05-11T11:19:19.425822-05:00 NW-CORE.jubacho.org rsyslogd: [origin software="rsyslogd" swVersion="8.2102.0" x-pid="22375" x-info="https://www.rsyslog.com"] exiting on signal 15.
2023-05-11T11:19:19.490017-05:00 NW-CORE.jubacho.org rsyslogd: imuxsock: Acquired UNIX socket '/run/systemd/journal/syslog' (fd 3) from systemd.  [v8.2102.0]
2023-05-11T11:19:19.490294-05:00 NW-CORE.jubacho.org rsyslogd: [origin software="rsyslogd" swVersion="8.2102.0" x-pid="22388" x-info="https://www.rsyslog.com"] start
```

So if the log file is actually 0 and `logging persistent` is enabled, it means there's a problem.